report_xml/controllers/main.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+import json
+import time
+
+from odoo.addons.web.controllers import main as report
+from odoo.http import content_disposition, request, route, serialize_exception
+from odoo.tools import safe_eval, html_escape
+from werkzeug.urls import url_decode
+
+
+class ReportController(report.ReportController):
+    @route()
+    def report_routes(self, reportname, docids=None, converter=None, **data):
+        if converter == 'xml':
+            report = request.env['ir.actions.report']._get_report_from_name(
+                reportname)
+            context = dict(request.env.context)
+
+            if docids:
+                docids = [int(i) for i in docids.split(',')]
+            if data.get('options'):
+                data.update(json.loads(data.pop('options')))
+            if data.get('context'):
+                # Ignore 'lang' here, because the context in data is the one
+                # from the webclient *but* if the user explicitely wants to
+                # change the lang, this mechanism overwrites it.
+                data['context'] = json.loads(data['context'])
+                if data['context'].get('lang'):
+                    del data['context']['lang']
+                context.update(data['context'])
+
+            xml = report.with_context(context).render_qweb_xml(docids,
+                                                               data=data)[0]
+            xmlhttpheaders = [('Content-Type', 'text/xml'),
+                              ('Content-Length', len(xml))]
+            return request.make_response(xml, headers=xmlhttpheaders)
+        else:
+            return super(ReportController, self).report_routes(
+                reportname, docids, converter, **data)
+
+    @route()
+    def report_download(self, data, token):
+        requestcontent = json.loads(data)
+        url, report_type = requestcontent[0], requestcontent[1]
+        if report_type == 'qweb-xml':
+            try:
+                converter = extension = 'xml'
```

Comment:
Does a better reading make up for this? maybe better to think of a more generic variable name instead of use two for the same value?